py_zipkin/storage.py

```diff
@@ -195,3 +225,35 @@ def set_default_tracer(tracer):
         return _set_contextvars_tracer(tracer)
 
     return _set_thread_local_tracer(tracer)
+
+
+class TracingThread(threading.Thread):
+    """A tracing-aware Thread subclass.
+
+    When run, if the parent thread had a Tracer, this thread gets a copy
+    of its Tracer.  The copy has the properties that stored spans will be
+    available in the parent thread, but the context stack is a deep copy (i.e.
+    contexts pushed or popped in this thread will not affect the parent
+    thread's Tracer's context stack).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(TracingThread, self).__init__(*args, **kwargs)
+        self.__orig_tracer = None
+        if has_default_tracer():
+            self.__orig_tracer = get_default_tracer().copy()
```

Comment:
It's arguable that the `.copy()` should take place in `.start()` in case the parent thread adds more spans between the time this thread is instantiated and when it's started.

In practice, that seems unlikely, but looking at this again, I think doing the `get_default_tracer().copy()` in `TracingThread.start()` would be better.

Reply:
Fixed this issue with a follow-up commit.